### PR TITLE
Add Restate CLI docs

### LINKED ISSUE #248: CLI page with info on installation, configuration and a few key commands

### PR DESCRIPTION
**Pull request by gvdongen**
Fixes #248 
- CLI overview page
- Describes installation 
- Describes configuration
- Describes a few key commands

### CODE REVIEW

File: docs/restate/cli.mdx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+restate example
+```
+
+Follow the prompts to select which template you want to download.
+
+**Listing your services**
+
+```shell
+restate svc list
+```
```
Comment (AhmedSoliman) [inline docs/restate/cli.mdx:87]: A very useful/powerful command is also `restate svc status` which can take an optional argument (service name) if you want to filter out the output by a specific service. 

It shows running the summary of the service and counters for invocations in various states. Is also shows the a sample of service keys that has active ongoing invocations.

File: docs/restate/cli.mdx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+There are two ways to configure the CLI: via environment variables or via a configuration file.
+
+### Using environment variables
+
+You can specify the following environment variables:
+
+- `RESTATE_HOST`: The hostname/IP address of the server. Default is `localhost`.
+- `RESTATE_HOST_SCHEME`: Default is `http`.
+- `RESTATE_META_URL`: To specify the full URL of meta service (scheme+host+port).
+- `RESTATE_DATAFUSION_HTTP_URL`: To specify the full URL of datafusion HTTP endpoint.
```
Comment (AhmedSoliman) [inline docs/restate/cli.mdx:46]: I'm very hesitant about exposing this as a "data fusion" thing. I would like to combine this with the META URL under one name "admin" URL. Let me work on this first.

File: docs/restate/cli.mdx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+For example, to connect to a Restate meta running at `http://myhost:7777`, save the following in a `.env` file:
+
+```
+RESTATE_META_URL=http://myhost:7777
+```
+
+## Usage
+
+Here are a few useful commands:
```
Comment (AhmedSoliman) [inline docs/restate/cli.mdx:67]: Suggested: `restate whoami` is a good way to know if your CLI is configured correctly or not.

File: docs/restate/cli.mdx (hunk 1 above)
Comment (gvdongen) [reply docs/restate/cli.mdx:87]: Cool, that is indeed a useful one! I added it.

**Comment by AhmedSoliman** [comment]
This is great and it would be very nice if we can add some screenshots of example outputs.

**Comment by gvdongen** [comment]
> This is great and it would be very nice if we can add some screenshots of example outputs.

I think it would be best to do this when the CLI and its printed output are a bit more stable since this will be intensive to maintain. Is that okay?

**Pull request merged**